.azure-pipelines/templates/jobs/packaging-jobs.yml

```diff
@@ -63,22 +63,47 @@ jobs:
       matrix:
         amd64:
           ARCH: amd64
-        arm64:
-          ARCH: arm64
-        armhf:
-          ARCH: armhf
+        # Do not run the QEMU jobs for the full-test-suite pipeline
+        ${{ if ne(variables['System.DefinitionId'], 4 )}}:
+          arm64:
+            ARCH: arm64
+          armhf:
+            ARCH: armhf
     pool:
       vmImage: ubuntu-18.04
     steps:
       - script: |
-          snap/local/build.sh ${ARCH}
+          tools/snap/build.sh ${ARCH}
           mv *.snap $(Build.ArtifactStagingDirectory)
         displayName: Build Certbot snap
       - task: PublishPipelineArtifact@1
         inputs:
           path: $(Build.ArtifactStagingDirectory)
           artifact: snap-$(arch)
         displayName: Store snap artifact
+  - job: snap_dns_build
+    strategy:
+      matrix:
+        amd64:
+          ARCH: amd64
+        # Do not run the QEMU jobs for the full-test-suite pipeline
+        ${{ if ne(variables['System.DefinitionId'], 4 )}}:
+          arm64:
+            ARCH: arm64
+          armhf:
+            ARCH: armhf
+    pool:
+      vmImage: ubuntu-18.04
+    steps:
+      - script: |
+          tools/snap/build_dns.sh ${ARCH} ALL
```

Comment:
You said this approach is faster than building the DNS snaps in parallel because you don't have to duplicate the setup for each snap. Did you get a chance to test this? Is this also true when building the snaps for ARM?

Reply:
Yes I tested it. This is particularly critical on snaps for ARM. For these architectures, it takes between 4 and 5 minutes to get the system ready for actually starting the snap build. So we would add here globally more than 2 hours of build for the two ARM architectures. For `amd64` it takes a minute. Less significant, but still.

I think it is the best balance for QEMU-based job. On remote builds maybe we could build in parallel, if the number of parallel worker we can use is sufficient.

Reply:
Have a link to the build? It surprises me a bit because sure it adds 2 hours of work, but that work can be done in parallel rather than 40 minutes of work in series.

Reply:
I fact I did not implement a parallel build of each DNS snap. I used the approach you see here, and measured the time it takes to do the setup on a Azure job. I can try a parallel call to see. However if we go that way, we will need to explicitly set the list of dns plugins to build into the pipeline definition.

Reply:
Having to explicitly list the the DNS plugins in the pipeline doesn't seem that bad to me. I don't think we've added one in over 2 years and I don't think we ever will add them like we were in the past.

With that said, I think we could potentially generate this list dynamically if we wanted based on the example of building the matrix from a variable containing JSON at https://docs.microsoft.com/en-us/azure/devops/pipelines/process/phases?tabs=yaml&view=azure-devops#multi-job-configuration.

It looks like you're currently testing this so we'll see what you find, but I personally suspect that with the current setup, the fastest approach would be building all of the plugins for amd64 in one job and building all of the plugins for ARM in parallel. (We could maybe get even faster by not giving each ARM build its own job and building a few in each job, but I think this is probably more trouble than it's worth.) I think that if we switch to using `snapcraft remote-build`, it would be fastest to build each DNS plugin in its own job, but to build the DNS plugin for all architectures in that job.

Reply:
Ok here are the results of my tests. I ran two pipelines.
* one with the current approach where all dns plugins build sequentially https://dev.azure.com/certbot/certbot/_build/results?buildId=2291&view=results
* one with an approach where all dns plugins are build in parallel https://dev.azure.com/certbot/certbot/_build/results?buildId=2290&view=results

The sequential build ran for 46min30s, but it should have been 42min, because the `snap_dns_run` stale unexpectedly during 4min during the installation of the dependencies. The parallel build ran for 53min30s, but it should have been 55min, because `snap_dns_run` was not executed.

So the sequential approach is faster (10min less), leaving 8 executors available for other pipelines after 35min of build, and avoid to add complexity to the pipeline to split the jobs.